SQL_Assignment_3.sql

```diff
@@ -0,0 +1,258 @@
+SELECT DATALENGTH('') AS BYTELENGTH;
+--EMPTY STRING is valid but empty sequence
+
+
+--11. Find the position of 'Server' in the string 'Microsoft SQL Server'.
+SELECT CHARINDEX('Server','Microsoft SQL Server') AS STARTINDEX;
+
+--12. From an 'Emails' table, write a query to extract the domain name from email addresses.
+SELECT SUBSTRING(EMAIL, CHARINDEX('@',EMAIL)+1,LEN(EMAIL))
+FROM EMAILEMP;
+
+--using reverese and substring function
+SELECT REVERSE(SUBSTRING(REVERSE(EMAIL), 1, CHARINDEX('@', REVERSE(EMAIL)) - 1)) AS DOMAIN
+FROM EMAILEMP;
+
+
+--13. Find the position of the first number in the string 'ABC123DEF456'.
+SELECT PATINDEX('%[0-9]%','ABC123DEF456') AS FIRST_NUMBER;
+
+--14. Write a query to find all product names from a 'Products' table that contain a number.
+SELECT PRODUCT_NAME
+FROM PRODUCT 
+WHERE PATINDEX('%[0-9]%', PRODUCT_NAME) > 0;
+
+--15. Join the strings 'SQL', 'Server', and '2022' with spaces between them.
+SELECT 'SQL'+' '+'Server'+' '+' '+ '2022' AS CONCATINATEDSTRING;
+
+--16. From 'Employees' table with 'FirstName' and 'LastName' columns, create a 'FullName' column.
+SELECT CONCAT(FIRSTNAME, ' ', LASTNAME) AS FULLNAME FROM Employees ;
+
+
+--17. Join the array C with a hyphen as the separator.
+SELECT CONCAT('SQL','-', 'Server','-', '2022') AS JOINEDSTRING;
+--18. From an 'Addresses' table, combine 'Street', 'City', 'State', and 'ZIP' columns into a single address string.
+SELECT CONCAT(Street,', ', City,', ', State,', ',  ZIP) AS ADDRESS FROM Addresses;
+
+--19. Change all occurrences of 'a' to 'e' in the string 'database management'.
+SELECT REPLACE('database management','a','e');
+--20. From a 'Products' table, write a query to replace all spaces in product names with underscores.
+SELECT REPLACE(PRODUCT_NAME,' ','-') FROM PRODUCT;
+
+--21. Create a string of 10 asterisks (*).
+SELECT REPLICATE('*',10);
+
+
+--22. Write a query to pad all product codes in a 'Products' table to a length of 10 characters with leading zeros.
+SELECT RIGHT(REPLICATE('0',10)+PRODUCT_ID,10) AS PRODCUT_ID_10 FROM PRODUCT;
+
+--23. Insert the string 'New ' at the beginning of 'York City'.
+UUPDATE CUSTOMERS
+SET CITY = CONCAT('New ', CITY)
+WHERE CITY = 'York City';
+
+SELECT * FROM CUSTOMERS;
+--24. From an 'Emails' table, mask the username part of email addresses, showing only the first and last characters.
+SELECT * FROM EMAILEMP;
+SELECT CONCAT(LEFT(EMAIL,1),REPLICATE('*',LEN(EMAIL)-2),RIGHT(EMAIL,1)) AS MASKED 
+FROM EMAILEMP;
+
+--25. Convert the string 'sql server' to uppercase.
+SELECT UPPER('sql server');
+--26. Write a query to convert all customer names in a 'Customers' table to uppercase.
+SELECT UPPER(CUSTOMERNAME)AS CUSTOMERNAME FROM CUSTOMERS;
+
+--27. Convert the string 'SQL SERVER' to lowercase.
+SELECT LOWER('SQL SERVER') AS LOWER_CASE;
+--28. From a 'Products' table, write a query to convert all product descriptions to lowercase.
+SELECT LOWER(PRODUCT_ID) FROM PRODUCT;
```

Comment:
Should be PRODUCT_NAME instead of PRODUCT_ID.

Reply:
Corrections made